src/pyOpenMS/CMakeLists.txt

```diff
@@ -385,23 +385,37 @@ configure_file(${_env_py_in} ${_env_py} @ONLY)
 #------------------------------------------------------------------------------
 # create targets in makefile
 
+# Fix race condition: Use a single sentinel file as OUTPUT instead of multiple files
+# This ensures the custom command runs only once even with parallel make
+set(PYGEN_SENTINEL "${CMAKE_BINARY_DIR}/pyOpenMS/.cpp_extension_generated")
+
 IF(LINUX AND PY_NO_OUTPUT)
   add_custom_command(
-    OUTPUT ${PYCPPFILES} ${PYPYXFILES}
+    OUTPUT ${PYGEN_SENTINEL}
     DEPENDS OpenMS # fake dependency to see if a header file has changed (TODO should probably depend on pxds)
     COMMAND ${Python_EXECUTABLE} create_cpp_extension.py 2> /dev/null
+    COMMAND ${CMAKE_COMMAND} -E touch ${PYGEN_SENTINEL}
     COMMENT "Creating C++ extension with autowrap and Cython"
     WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/pyOpenMS )
 ELSE()
   add_custom_command(
-    OUTPUT ${PYCPPFILES} ${PYPYXFILES}
+    OUTPUT ${PYGEN_SENTINEL}
     DEPENDS OpenMS # fake dependency to see if a header file has changed (TODO should probably depend on pxds)
     COMMAND ${Python_EXECUTABLE} create_cpp_extension.py
+    COMMAND ${CMAKE_COMMAND} -E touch ${PYGEN_SENTINEL}
     COMMENT "Creating C++ extension with autowrap and Cython"
     WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/pyOpenMS )
 ENDIF()
 
+# Ensure all generated files depend on the sentinel
+add_custom_command(
+  OUTPUT ${PYCPPFILES} ${PYPYXFILES}
+  DEPENDS ${PYGEN_SENTINEL}
+  COMMENT "Generated files depend on successful code generation"
+)
+
 # this acts as a mutex in make, to not have race conditions on the above command in the next parallel targets
+# USES_TERMINAL ensures the command runs in a single terminal session, preventing parallel execution
```

Comment:
?

Reply:
And what about the USES_TERMINAL Connect?

Reply:
Hmm, that's vestigial from debugging. I'll remove it.